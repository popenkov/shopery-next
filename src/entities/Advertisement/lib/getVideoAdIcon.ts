import { LeafIcon, PriceTagIcon } from '@/shared/ui/icons';

const icons = {
  PriceTagIcon: PriceTagIcon,
  LeafIcon: LeafIcon,
};

export const getVideoAdIcon = (iconName: keyof typeof icons) => {
  return icons[iconName];
};
