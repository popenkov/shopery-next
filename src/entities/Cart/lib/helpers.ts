import { CartSchema } from '../model/types/cart-schema';

export const changeItemsAmountInCart = (state: CartSchema, id: string, amount: number) => {
  const newState: CartSchema = JSON.parse(JSON.stringify(state));
  const findItemIndex = newState?.cart?.findIndex((item) => item.id === id);
  newState.cart[findItemIndex].amount = amount;
  newState.totalPrice = getTotalPrice(newState);
  newState.totalAmount = getTotalAmount(newState);
  return newState;
};

// надо вернуть объект
export const getTotalPrice = (state: CartSchema) => {
  return state.cart.reduce(
    (acc, current) => {
      acc.USD += current.price.USD * current.amount;
      acc.EUR += current.price.EUR * current.amount;
      return acc;
    },
    { USD: 0, EUR: 0 },
  );
};

export const getTotalAmount = (state: CartSchema) => {
  return state.cart.reduce((acc, item) => {
    return acc + item.amount;
  }, 0);
};
