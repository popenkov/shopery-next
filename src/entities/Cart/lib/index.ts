export { changeItemsAmountInCart, getTotalPrice, getTotalAmount } from './helpers';
