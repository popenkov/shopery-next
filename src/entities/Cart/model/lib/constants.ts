import { CartSchema } from '../types/cart-schema';

export const getTotalPrice = (state: CartSchema) => {
  return state.cart.reduce((acc, item) => {
    return acc + item.price * item.amount;
  }, 0);
};

export const getTotalAmount = (state: CartSchema) => {
  return state.cart.reduce((acc, item) => {
    return acc + item.amount;
  }, 0);
};
