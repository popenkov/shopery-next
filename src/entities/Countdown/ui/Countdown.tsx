'use client';

import React, { DetailedHTMLProps, FC, HTMLAttributes } from 'react';

import cn from 'classnames';


import { useCountDown } from '@/shared/lib/hooks/useCountDown/useCountDown';

import cls from './Countdown.module.scss';

interface Props
    extends DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
    date: string;
    className?: string;
}

export const Countdown: FC<Props> = ({ date, className }) => {
    const { days, hours, minutes, seconds } = useCountDown(date);
    return (
        <div className={cn(cls.counter, className)}>
            <div className={cls.counterItem}>
                <span className={cls.counterValue}>{days}</span>
                <span className={cls.counterDescription}>Days</span>
            </div>
            <span className={cls.counterSeparator}>:</span>
            <div className={cls.counterItem}>
                <span className={cls.counterValue}>{hours}</span>
                <span className={cls.counterDescription}>Hours</span>
            </div>
            <span className={cls.counterSeparator}>:</span>
            <div className={cls.counterItem}>
                <span className={cls.counterValue}>{minutes}</span>
                <span className={cls.counterDescription}>Mins</span>
            </div>
            <span className={cls.counterSeparator}>:</span>
            <div className={cls.counterItem}>
                <span className={cls.counterValue}>{seconds}</span>
                <span className={cls.counterDescription}>Secs</span>
            </div>
        </div>
    );
};
