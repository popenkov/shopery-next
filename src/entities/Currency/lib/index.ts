export * from './constants';
export { getUserCurrency, setUserCurrency } from './setCurrency';
