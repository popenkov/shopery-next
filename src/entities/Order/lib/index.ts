export const ORDERS_LOCALSTORAGE_KEY = 'orders';
