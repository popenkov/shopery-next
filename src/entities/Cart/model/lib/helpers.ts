import { CartSchema } from '../types/cart-schema';

import { getTotalAmount, getTotalPrice } from './constants';

export const changeItemsAmountInCart = (state: CartSchema, id: string, amount: number) => {
  const newState: CartSchema = JSON.parse(JSON.stringify(state));
  const findItemIndex = newState?.cart?.findIndex((item) => item.id === id);
  newState.cart[findItemIndex].amount = amount;
  newState.totalPrice = getTotalPrice(newState);
  newState.totalAmount = getTotalAmount(newState);
  return newState;
};
