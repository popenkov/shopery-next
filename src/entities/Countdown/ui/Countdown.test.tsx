import React from 'react';
import { render } from '@testing-library/react';

import * as hooks from '@/shared/lib/hooks';

import { Countdown } from './Countdown';

jest.mock('@/shared/lib/hooks', () => ({
  useCountDown: jest.fn(() => ({
    days: 10,
    hours: 5,
    minutes: 30,
    seconds: 15,
  })),
}));

describe('Countdown component', () => {
  it('renders correctly', () => {
    const { container } = render(<Countdown date="2024-03-16T14:30:00.000Z" />);
    expect(container).toMatchSnapshot();
  });

  it('renders days, hours, minutes and seconds', () => {
    const { getByText } = render(<Countdown date="2024-03-16T14:30:00.000Z" />);
    expect(getByText('10')).toBeInTheDocument();
    expect(getByText('5')).toBeInTheDocument();
    expect(getByText('30')).toBeInTheDocument();
    expect(getByText('15')).toBeInTheDocument();
  });

  it('renders descriptions for days, hours, minutes and seconds', () => {
    const { getByText } = render(<Countdown date="2024-03-16T14:30:00.000Z" />);
    expect(getByText('Days')).toBeInTheDocument();
    expect(getByText('Hours')).toBeInTheDocument();
    expect(getByText('Mins')).toBeInTheDocument();
    expect(getByText('Secs')).toBeInTheDocument();
  });

  it('renders separators between values', () => {
    const { getAllByText } = render(<Countdown date="2024-03-16T14:30:00.000Z" />);
    const separators = getAllByText(':');
    expect(separators).toHaveLength(3);
  });

  it('calls useCountDown hook with correct date', () => {
    const useCountDownSpy = jest.spyOn(hooks, 'useCountDown');
    render(<Countdown date="2024-03-16T14:30:00.000Z" />);
    expect(useCountDownSpy).toHaveBeenCalledTimes(1);
    expect(useCountDownSpy).toHaveBeenCalledWith('2024-03-16T14:30:00.000Z');
  });
});
