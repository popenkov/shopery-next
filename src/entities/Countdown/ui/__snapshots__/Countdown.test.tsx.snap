// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countdown component renders correctly 1`] = `
<div>
  <div
    class="counter"
  >
    <div
      class="counterItem"
    >
      <p
        class="text counterValue primary heading_5 left regular"
      >
        10
      </p>
      <span
        class="counterDescription"
      >
        Days
      </span>
    </div>
    <p
      class="text counterSeparator primary body_xl left regular"
    >
      :
    </p>
    <div
      class="counterItem"
    >
      <p
        class="text counterValue primary heading_5 left regular"
      >
        5
      </p>
      <span
        class="counterDescription"
      >
        Hours
      </span>
    </div>
    <p
      class="text counterSeparator primary body_xl left regular"
    >
      :
    </p>
    <div
      class="counterItem"
    >
      <p
        class="text counterValue primary heading_5 left regular"
      >
        30
      </p>
      <span
        class="counterDescription"
      >
        Mins
      </span>
    </div>
    <p
      class="text counterSeparator primary body_xl left regular"
    >
      :
    </p>
    <div
      class="counterItem"
    >
      <p
        class="text counterValue primary heading_5 left regular"
      >
        15
      </p>
      <span
        class="counterDescription"
      >
        Secs
      </span>
    </div>
  </div>
</div>
`;
